packages/go_router/lib/src/route_data.dart

```diff
@@ -11,12 +11,29 @@ import 'package:meta/meta_meta.dart';
 import 'route.dart';
 import 'state.dart';
 
+/// {@template route_data}
+/// A superclass for each route data
+/// {@endtemplate}
+abstract class RouteData {
+  /// {@macro route_data}
+  const RouteData();
+
+  /// [navigatorKey] is used to point to a certain navigator
+  /// or pass it into the shell route
+  ///
+  /// In case of [ShellRoute] it will instantiate a new navigator
+  /// with the given key
+  ///
+  /// In case of [GoRoute] it will use the given key to find the navigator
+  GlobalKey<NavigatorState>? get navigatorKey => null;
```

Comment:
I think it is better to keep it as per subclass for now. With the introduction of StatefulShellRoute, it may have multiple navigatorkeys

Reply:
then I will try to familiarise myself with that PR first

Reply:
I think the best course of action will be to not having a base implementation for now. The base class will just be an empty class for typing purposes.

Reply:
this seems to be not addressed yet?